Makefile

```diff
@@ -78,7 +88,7 @@ run: welcome githooks-commit install build
 	@$(NODE) build/bundle-$(CALYPSO_ENV).js
 
 node-version:
-	@NPM_GLOBAL_ROOT=$(shell $(NPM) -g root) $(BIN)/check-node-version
+	@$(BIN)/check-node-version
```

Comment:
Loading semver this way is unorthodox, yes, but it is well-explained in the comments and doesn't seem to be causing any problems.  A PR to do a simple version compare would be fine, but we shouldn't use string comparison for this, at the very least we'd want to get rid of the leading `v` and compare each component between `.`s.  Adding in the extra code to do this doesn't seem like a clear win to me over the current approach.

The goal of this check is to quickly execute at the beginning of the build process and abort if continuing is a waste of time.  This should be done before `npm install` since that takes a while due to all of our dependencies.  And as far as `npm install` flagging based on the `engines.node` property...

> Note that, unless the user has set the engine-strict config flag, this field is advisory only. --[npm docs](https://docs.npmjs.com/files/package.json#engines)

(Somewhat ironically, the `make node-version` check uses `process.exitCode` from Node 0.12.x, so it doesn't actually abort the build for older Node versions.  PR to fix this at #3581.)